pkg/apis/logging/v1/output_types.go

```diff
@@ -0,0 +1,109 @@
+package v1
+
+import (
+	"github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1/outputs"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// Output defines a destination for log messages.
+type Output struct {
+	// Name used to refer to the output from a `pipeline`.
+	//
+	// +required
+	Name string `json:"name"`
+
+	// OutputType is inlined with a required `type` and optional extra configuration.
+	OutputType `json:",inline"`
+
+	// Destination of log messages, an absolute URL (must have a scheme).
+	//
+	// Valid URL schemes depend on `type`.  Special schemes 'tcp', 'udp' and
+	// 'tls' are used for output types that don't define their own URL scheme.
+	// Example:
+	//
+	//     { type: syslog, destination: tls://syslog.example.com:1234 }
+	//
+	// TLS with server authentication is enabled by the URL scheme, for
+	// example 'tls' or 'https'.  See `secretRef` for TLS client authentication.
+	//
+	// +required
+	Destination string `json:"destination"`
+
+	// SecretRef refers to a `secret` object for secure communication.
+	//
+	// Client-authenticated TLS is enabled if the secret contains keys
+	// `tls.crt`, `tls.key` and `ca.crt`. Output types with password
+	// authentication will use keys `password` and `username`, not
+	// the exposed 'username@password' part of the `destination` URL.
+	//
+	// +optional
+	SecretRef corev1.SecretReference `json:"secret,omitempty"`
+
+	// Insecure must be true for intentionally insecure outputs.
+	// Has no function other than a marker to help avoid configuration mistakes.
+	//
+	// +optional
+	Insecure bool `json:"insecure,omitempty"`
+
+	// Reconnect configures how the output handles connection failures.
+	// Auto-reconnect is enabled by default.
+	//
+	// +optional
+	Reconnect *Reconnect `json:"reconnect,omitempty"`
```

Comment:
Consider omitting if we are unable to support initially with the GA release

Reply:
+1